subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -55,12 +56,73 @@ public Download(Logger logger, DownloadProgressListener progressListener, String
         configureProxyAuthentication();
     }
 
-    private void configureProxyAuthentication() {
-        if (System.getProperty("http.proxyUser") != null) {
-            Authenticator.setDefault(new ProxyAuthenticator());
+    private void populateProxyPropertiesFromEnvironmentVariable(String protocol) {
+        // Do not fallback to environment variables if a proxy has been set via properties.
+        if (System.getProperty(protocol + ".proxyHost") != null) {
+            return;
+        }
+
+        String variableName = protocol + "_proxy";
+        String proxyString = System.getenv(variableName);
+        if (proxyString == null) {
+            proxyString = System.getenv(variableName.toUpperCase());
+        }
+
+        try {
+            URL proxyUrl = new URL(proxyString);
```

Comment:
I'll fix this later as I decided to drop that commit from this PR for simplicity. Because I would need to add support for `http(s)_proxy` environment variables also to the main Gradle, not just the wrapper.